source/wp-content/plugins/handbook/inc/table-of-contents.php

```diff
@@ -164,7 +164,7 @@ public function add_toc( $content ) {
 
 		$toc .= "</ul>\n</div>\n";
 
-		return $toc . $content;
+		return array('toc' => $toc, 'content' => $content);
```

Comment:
My suggestion would be to rename this from `add_toc` to something like `parse_content`, then add a new `add_toc` which is similar to:
```
function add_toc( $content ) {
   $parts = $this->parse_content( $content );
   return $parts['toc'] . $parts['content'];
}
```

That way the function syntax remains unchanged for other uses of the plugin.. and for this filter in the plugin: `add_filter( 'the_content', array( $this, 'add_toc' ) );` which would expect a stringy return.

Reply:
Done! Thanks for the suggestion. Tested that this works with the old `content-reference.php`